src/server/services/data-transform.service.test.js

```diff
@@ -563,6 +564,48 @@ describe("data-transform.service transformAnswersForSummary()", () => {
   });
 });
 
+describe("data-transform.service trimUprn()", () => {
+  let result;
+
+  it("Should return empty string when UPRN is not a string or is empty string", () => {
+    const badText = [[], {}, null, undefined, ''];
+    
+    badText.forEach(text => {
+      const result = trimUprn(text);
+      expect(result).toBe('');
+    });
+  });
+
+  it("Should return empty string when UPRN is completely invalid", () => {
+    const badText = ['asdasd333333', 'dsdfsdfs344', '%f', '--4'];
+    
+    badText.forEach(text => {
+      const result = trimUprn(text);
+      expect(result).toBe('');
+    });
+  });
+
+  it("Should return UPRN trimmed of trailing non-numeric characters when present", () => {
+    const uprn = ['12334sdfsd', '1233456789-1', '9999999%%%dd3'];
```

Comment:
const uprn is never used.  Is this line needed?
Could use uprn[0] and so on below it.

Reply:
Good spot. I meant to remove it to be honest. This test requires more specific validation so looping through isn't as useful. I suppose I could have an array of objects with input and output in but will just take line out.